draft-irtf-cfrg-hpke.md

```diff
@@ -830,16 +890,56 @@ security models and assumptions, and do not consider attackers capable of quantu
 computation. A full proof of post-quantum security would need to take this
 difference into account, in addition to simply using a post-quantum KEM.
 
-## Domain Separation
-
-\[\[RFC editor: please change "RFCXXXX" to the correct number before
-publication.]]
-
-The KeySchedule procedure includes the domain separation string "RFCXXXX" in
-each Expand invocation. This ensures any secrets derived in HPKE are independent
-from those used in other protocols, even when derived from the same IKM (secret).
-Derivation of the KeySchedule 'secret' does not include domain separation as it
-is an intermediate value not exposed by the protocol.
+## Pre-Shared Key Recommendations {#security-psk}
+
+In mode PSK and mode AuthPSK, the PSK SHOULD be of length Nh bytes or
+longer, and SHOULD have Nh bytes of entropy or more. Using a PSK shorter
+than Nh bytes is permitted. A PSK that is longer than Nh bytes or that
+has more than Nh bytes of entropy, respectively, does not increase the
+security level of HPKE, because the extraction step involving the PSK
+only outputs Nh bytes.
+
+HPKE is specified to use HKDF as key derivation function. HKDF is not
+designed to slow down dictionary attacks, see {{?RFC5869}}. Thus,
+HPKE's PSK mechanism is not suitable for use with a low-entropy
+password as the PSK: in scenarios in which the adversary knows the
+KEM shared secret zz and has access to an oracle for decryption success
+of an HPKE ciphertext, it can perform a dictionary attack on the PSK.
+If an application does not use Seal to create HPKE ciphertexts but only
+the Export interface, the adversary can rely on other recipient behavior
+which is observably different when using a wrong key.
+
+Scenarios in which the adversary knows the KEM shared secret zz
+depend on the KEM. In the case of DHKEM, it is enough if the adversary
+knows all private keys of one participant. 
```

Comment:
What do you mean by "it is enough"?  Also, I don't understand the dependency on DHKEM here.

Reply:
The adversary knows the KEM shared secret zz in certain compromise scenarios. We were hesitant to make a general claim about compromise scenarios with a general KEM. But maybe it's actually possible. Have to think about it.

The “enough” because people could think that for DHKEM's AuthPSK it is not enough to know only the recipient's private static key, but that knowing the sender's private static key is also necessary; because of key-compromise impersonation this is not the case. We could do without the “enough” and say “For DHKEM this is the case if the adversary knows all private keys of one participant”.

Reply:
For a “good or bad PSK” oracle, knowing the secrets of _one_ party is enough because: Each one of the two parties involved in HPKE must be able to decrypt a message, and each party only knows its own secrets.

I will rephrase the paragraph accordingly.

Reply:
I see.  I think I misunderstood the "Scenaris where the adversary knows zz" above.  With your new text above, it's clearer.

Reply:
To everyone: I attempted to clarify this in commit https://github.com/cfrg/draft-irtf-cfrg-hpke/pull/50/commits/1c84680727de94679dcf1f96c7c401a0cc8178b8, please let me know if you think this works. I decided to attach this to the previous paragraph, because it kind of belongs to it; please let me know if you like that.

Reply:
This reads well to me. Thanks for clarifying, @blipp!